File: 프로그래머스/2/178870. 연속된 부분 수열의 합/연속된 부분 수열의 합.js
function solution(sequence, k) {
    let left = 0;
    let right = 0;
    let sum = sequence[0];
    let answer = [];

    while (right < sequence.length) {
        if (sum === k) {
            answer.push([left, right]);
            sum -= sequence[left];
            left++;
        } else if (sum < k) {
            right++;
            if (right < sequence.length) sum += sequence[right];
        } else { 
            sum -= sequence[left];
            left++;
        }
    }

    answer.sort((a, b) => {
        const lenA = a[1] - a[0];
        const lenB = b[1] - b[0];
        if (lenA === lenB) return a[0] - b[0];
        return lenA - lenB;
    });

    return answer[0];
}

File: 프로그래머스/2/178870. 연속된 부분 수열의 합/연속된 부분 수열의 합.py
def solution(sequence, k):
    N = len(sequence)
    answer = [0, N-1] 
    left, right, total = 0, 0, 0
    
    while right < N:
        total += sequence[right]
        while total > k:
            total -= sequence[left]
            left += 1
        if total == k:
            if right - left < answer[1] - answer[0]:
                answer = [left, right]
        right += 1
    return answer